### PR TITLE
Config group

### PR DESCRIPTION
**Pull request by SoumyaLakshmir**
Committing Updated changes in Tags.

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,8 +1,8 @@
 server.port=8081
 server.servlet.context-path=/sourcecourse-config/api
-spring.datasource.url=jdbc:mysql://localhost:3306/sourcecourse?useSSL =true
+spring.datasource.url=jdbc:mysql://localhost:3306/sourcecource?useSSL =true
```
Comment (shijukbabu-ust) [inline src/main/resources/application.properties:3]: revert this change & commit

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,8 +1,8 @@
 server.port=8081
```
Comment (shijukbabu-ust) [inline src/main/resources/application.properties:1]: revert this file changes and commit. 
Port, context-path and datasource url should be same as  dev branch.

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/configuration/controller/ProjectGroupController.java [hunk 3]
```diff
@@ -50,16 +70,85 @@ public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@
...
+	/**
+	 * 
+	 * @param tags
+	 * @param add  tags to project Group
+	 * @return
+	 */
+	@PostMapping("/{id}/tags")
+    public ResponseEntity<ProjectGroup> addTagToProjectGroup(@PathVariable ("id") Long uid, @RequestBody  List<String> tags) {
+        ProjectGroup updatedProjectGroup = projectGroupService.addTagToProjectGroup(uid, tags);
+        return ResponseEntity.ok(updatedProjectGroup);
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/configuration/controller/ProjectGroupController.java:108]: Do not return project group entity class. Either return ProjectGroupResponse or return only the tags

File: src/main/java/com/ust/sourcecourse/configuration/repository/ProjectGroupRepository.java [hunk 4]
```diff
@@ -9,4 +10,10 @@
 public interface ProjectGroupRepository extends JpaRepository<ProjectGroup, Long> {
 
 	List<ProjectGroup> findByProjectUid(Long uid);
+
+	List<ProjectGroup> findAllByTagsIn(List<String> tags);
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/configuration/repository/ProjectGroupRepository.java:14]: If we are not using these methods, remove it.

File: src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java [hunk 5]
```diff
@@ -112,4 +118,69 @@ public boolean deleteProjectGroup(Long uid) {
...
+	 */
+
+	public ProjectGroup addTagToProjectGroup(Long uid, List<String> tags) {
+		ProjectGroup projectGroup = projectGroupRepository.findById(uid)
+				.orElseThrow(() -> new ResourceNotFoundException("ProjectGroup", "id", uid));
+		List<String> tag = projectGroup.getTags();
+		if (tag == null) {
+			tag = new ArrayList<>();
+		}
+		tag.addAll(tags);
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java:135]: We need to handle duplicate. If same tag is in the request, we should not add again. 
A set can be used

File: src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java [hunk 6]
```diff
@@ -112,4 +118,69 @@ public boolean deleteProjectGroup(Long uid) {
...
+		projectGroup.setTags(tag);
+		return projectGroupRepository.save(projectGroup);
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java:137]: Return List<String> tags or the ProjectGroupResponse

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java (hunk 6 above)
Comment (SoumyaLakshmir) [reply src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java:137]: Resolved

File: src/main/java/com/ust/sourcecourse/configuration/repository/ProjectGroupRepository.java (hunk 4 above)
Comment (SoumyaLakshmir) [reply src/main/java/com/ust/sourcecourse/configuration/repository/ProjectGroupRepository.java:14]: Resolved

File: src/main/java/com/ust/sourcecourse/configuration/controller/ProjectGroupController.java (hunk 3 above)
Comment (SoumyaLakshmir) [reply src/main/java/com/ust/sourcecourse/configuration/controller/ProjectGroupController.java:108]: Resolved

File: src/main/resources/application.properties (hunk 2 above)
Comment (SoumyaLakshmir) [reply src/main/resources/application.properties:1]: Resolved

**Review by shijukbabu-ust — APPROVED** [review]
